sim/src/analytics.rs

```diff
@@ -324,7 +324,7 @@ impl Analytics {
         &self,
         now: Time,
         r: BusRouteID,
-    ) -> BTreeMap<BusStopID, Histogram<Duration>> {
+    ) -> impl Iterator<Item=(BusStopID, Histogram<Duration>)> {
```

Comment:
I'm not sure about the net benefit here; the two call-sites got more complicated. What's the motivation for this kind of refactor?

Reply:
I thought that `BTreeMap` is a particularly heavy data structure that I would not allocate lightly without need, thus I removed the `collect`.